directories/src/redux/epics/node-epics.js

```diff
@@ -143,9 +147,34 @@ const requestGenesisHashEpic = (action$, store) => {
             brokerNode.completeGenesisHashPoW(txid, trytesArray[0])
           ).map(({ data }) => {
             const { purchase: genesisHash, numberOfChunks } = data;
-            return nodeActions.addGenesisHash({ genesisHash, numberOfChunks });
-          })
+            return { genesisHash, numberOfChunks };
+          }).catch(error => {
+            console.log("GENESIS HASH COMPLETE ERROR", error);
+            return Observable.empty();
+          });
         )
+        .mergeMap(({ genesisHash, numberOfChunks }) => {
+          const datamap = Datamap.generate(genesisHash, numberOfChunks);
+          const addresses = _.values(datamap);
+          const randomNumber = AppUtils.randomNumber(addresses.length);
+          const randomSectorAddresses = addresses.slice(randomNumber);
+          return Observable.fromPromise(
+             iota.findTransactions(randomSectorAddresses)
+          ).map(({ transaction }) => {
+            const { bundles, addresses, tags, approvees } = transaction;
+            return { bundles, addreses, tags, approvees, genesisHash, numberOfChunks };
+          });
+        })
+        .mergeMap(({ bundles, addreses, tags, approvees, genesisHash, numberOfChunks}) => {
+          return Observable.fromPromise(
+             iota.getTrytes(bundles)
+          ).map(({ trytesArray }) => {
+            const transactionObject = iota.utils.TransactionObject(trytesArray);
```

Comment:
For this I think we'll want to implement the "lightTransactionObjects" (or whatever it's called) like Jet implemented for the downloads on webinterface.  No need to do this now but it's something we'll want to keep in mind.

Reply:
@AaronVasquez I think that's a good idea.  We could either make all of them separate packages or just have one biggish package of shared things that both webinterface and webnode use.  If we go that route, I think oyster-utils is a good name.  

Jared's PR is already in but this could be his next task.  He could also put the encryption and the iota files in there as well.

Reply:
In utils we will create oyster-utils and we will try to implement there "lightTransactionObjects"

Reply:
I didnt understand yesterday... now I understand, I know..ok